src/utilities/codemirror/vml-adaptation.ts

```diff
@@ -0,0 +1,114 @@
+import { type CompletionContext, type CompletionResult } from '@codemirror/autocomplete';
+import { syntaxTree } from '@codemirror/language';
+import type { ChannelDictionary, CommandDictionary, FswCommand, FswCommandArgument } from '@nasa-jpl/aerie-ampcs';
+import { getNearestAncestorNodeOfType } from '../sequence-editor/tree-utils';
+import { RULE_FUNCTION_NAME, RULE_ISSUE, RULE_STATEMENT, TOKEN_STRING_CONST } from './vml-constants';
+import { getArgumentPosition } from './vml-tree-utils';
+
+export function vmlAutoComplete(
+  _channelDictionary: ChannelDictionary | null,
+  commandDictionary: CommandDictionary | null,
+): (context: CompletionContext) => CompletionResult | null {
+  return (context: CompletionContext) => {
+    if (!commandDictionary) {
+      return null;
+    }
+
+    const tree = syntaxTree(context.state);
+    const nodeBefore = tree.resolveInner(context.pos, -1);
+    const nodeCurrent = tree.resolveInner(context.pos, 0);
+    if (nodeBefore.name === RULE_ISSUE) {
+      return {
+        from: context.pos,
+        options: commandDictionary.fswCommands.map((fswCommand: FswCommand) => ({
+          apply: getStemAndDefaultArguments(commandDictionary, fswCommand),
+          info: fswCommand.description,
+          label: fswCommand.stem,
+          section: 'Command',
+          type: 'function',
+        })),
+      };
+    } else if (nodeCurrent.name === TOKEN_STRING_CONST) {
+      // also show if before argument
+
+      const containingStatement = getNearestAncestorNodeOfType(nodeCurrent, [RULE_STATEMENT]);
+      if (containingStatement) {
+        const functionNameNode = containingStatement.firstChild?.getChild(RULE_FUNCTION_NAME);
+        if (functionNameNode) {
+          const stem = context.state.sliceDoc(functionNameNode.from, functionNameNode.to);
+          const cmdDef = commandDictionary.fswCommandMap[stem];
+          if (!cmdDef) {
+            return null;
+          }
+
+          const argPos = getArgumentPosition(nodeCurrent);
+          if (argPos === -1) {
+            return null;
+          }
+
+          const argDef = cmdDef.arguments[argPos];
+          if (!argDef || argDef.arg_type !== 'enum') {
+            return null;
+          }
+
+          const enumValues = commandDictionary.enumMap[argDef.enum_name].values;
+          return {
+            filter: false,
+            from: nodeCurrent.from,
+            options: enumValues.map(enumValue => ({
+              apply: `"${enumValue.symbol}"`,
+              label: `${enumValue.symbol} (${enumValue.numeric})`,
+              section: `${argDef.name} values`,
+              type: 'keyword',
+            })),
+            to: nodeCurrent.to,
+          };
+        }
+      }
+    }
+
+    return null;
+  };
+}
+
+function getStemAndDefaultArguments(commandDictionary: CommandDictionary, cmd: FswCommand): string {
+  let s = cmd.stem;
```

Comment:
pedantic: Can we use `stem` instead of `s`?

Reply:
addressed in https://github.com/NASA-AMMOS/aerie-ui/pull/1472/commits/04a1ae3e63362cd962e5c192b3acf67d19efbf87